cartridge/failover.lua

```diff
@@ -242,6 +243,82 @@ local function apply_config(mod)
     )
 end
 
+local function fencing_check()
+    if assert(vars.client):check_quorum() then
+        return true
+    end
+
+    local confapplier = require('cartridge.confapplier')
+    local replicaset_uuid = confapplier.get_replicaset_uuid()
+
+    local topology_cfg = confapplier.get_readonly('topology')
+
+    local leaders_order = topology.get_leaders_order(topology_cfg, replicaset_uuid)
+    for _, instance_uuid in ipairs(leaders_order) do
+        local server = assert(topology_cfg.servers[instance_uuid])
+        if not topology.not_disabled(instance_uuid, server) then
+            goto continue
+        end
+
+        local member = membership.get_member(server.uri)
+
+        if member ~= nil
+        and (member.status == 'alive')
+        and (member.payload.uuid == instance_uuid)
+        then
+            goto continue
+        end
+
+        do return false end
+        ::continue::
+    end
+
+    return true
+end
+
+local function fencing_watch()
+    log.info(
+        'Fencing activated (step %s, timeout %s)',
+        vars.options.FENCING_PAUSE, vars.options.FENCING_TIMEOUT
+    )
+
+    local deadline = fiber.clock() + vars.options.FENCING_TIMEOUT
+    repeat
+        fiber.sleep(vars.options.FENCING_PAUSE)
+
+        if (fencing_check()) then
+            deadline = fiber.clock() + vars.options.FENCING_TIMEOUT
+        end
+    until fiber.clock() > deadline
+
+    log.warn('Fencing triggered')
+    assert(accept_appointments({[box.info.cluster.uuid] = box.NULL}))
```

Comment:
Assert in fiber is useless - it kills a fiber and never printed. Let's replace it with explicit `log`.